tests/test_utils.py

```diff
@@ -20,3 +43,70 @@ def test_get_s2_evalscript(all_bands: bool, reflectance: bool) -> None:
     assert input_units_str in evalscript
     output_sample_type_str = 'sampleType: "FLOAT32"' if reflectance else 'sampleType: "UINT16"'
     assert output_sample_type_str in evalscript
+
+
+pytestmark = pytest.mark.sh_integration
+
+
+@pytest.mark.parametrize(
+    "test_input, expected",
+    [
+        ({"bbox": BBOX1, "time_interval": (dt.datetime(2021, 1, 1), dt.datetime(2021, 1, 1))}, []),
+        (
+            {"bbox": BBOX1, "time_interval": (dt.datetime(2021, 1, 1), dt.datetime(2021, 1, 10))},
+            [
+                dt.datetime(2021, 1, 4, 16, 39, 29),
+                dt.datetime(2021, 1, 4, 16, 39, 44),
+                dt.datetime(2021, 1, 9, 16, 39, 29),
+                dt.datetime(2021, 1, 9, 16, 39, 43),
+            ],
+        ),
+        (
+            {
+                "bbox": BBOX1,
+                "time_interval": (dt.datetime(2021, 1, 1), dt.datetime(2021, 1, 10)),
+                "time_difference": dt.timedelta(seconds=600),
+            },
+            [dt.datetime(2021, 1, 4, 16, 39, 29), dt.datetime(2021, 1, 9, 16, 39, 29)],
+        ),
+    ],
+)
+def test_get_timestemps(test_input: dict, config: SHConfig, expected: List[dt.datetime]) -> None:
+    timestemps = get_timestemps(**test_input, config=config)
+    assert len(timestemps) == len(expected)
+    assert all([val.replace(tzinfo=None) == exp for exp, val in zip(expected, timestemps)])
+
+
+@pytest.mark.parametrize(
+    "test_input, expected_shape",
+    [
+        ({"bbox": BBOX1, "timestamps": [dt.datetime(2021, 1, 4)], "size": (60, 81), "all_bands": True}, (1, 81, 60)),
+        (
+            {
+                "bbox": BBOX1,
+                "timestamps": [dt.datetime(2021, 1, 4, 16, 39, 29), dt.datetime(2021, 1, 9, 16, 39, 29)],
+                "size": (60, 81),
+                "all_bands": True,
+            },
+            (2, 81, 60),
+        ),
+        (
+            {
+                "bbox": BBOX1,
+                "timestamps": [dt.datetime(2021, 1, 4, 16, 39, 29), dt.datetime(2021, 1, 9, 16, 39, 29)],
+                "size": (60, 81),
+                "all_bands": False,
+            },
+            (2, 81, 60),
+        ),
+    ],
+)
+def test_download_bands_and_valid_data_mask(
+    test_input: dict, config: SHConfig, expected_shape: Tuple[int, int, int]
+) -> None:
+    bands, mask = download_bands_and_valid_data_mask(**test_input, config=config)
+    band_num = 10 if not test_input["all_bands"] else 13
```

Comment:
this should be IMO part of the `expected_shape`
Then you can do `mask.shape == expected_shape[:-1]`